channeldb/migration29/migration.go

```diff
@@ -0,0 +1,498 @@
+package migration29
+
+import (
+	"bytes"
+	"encoding/binary"
+	"fmt"
+	"math"
+	"sync"
+
+	mig24 "github.com/lightningnetwork/lnd/channeldb/migration24"
+	mig26 "github.com/lightningnetwork/lnd/channeldb/migration26"
+	mig "github.com/lightningnetwork/lnd/channeldb/migration_01_to_11"
+	"github.com/lightningnetwork/lnd/kvdb"
+)
+
+// recordsPerTx specifies the number of records to be migrated in each database
+// transaction. In the worst case, each old revocation log is 28,057 bytes.
+// 20,000 records would consume 0.56 GB of ram, which is feasible for a modern
+// machine.
+//
+// NOTE: we could've used more ram but it doesn't help with the speed of the
+// migration since the most of the CPU time is used for calculating the output
+// indexes.
+const recordsPerTx = 20_000
+
+// MigrateRevocationLog migrates the old revocation logs into the newer format
+// and deletes them once finished, with the deletion only happens once ALL the
+// old logs have been migrates.
+func MigrateRevocationLog(db kvdb.Backend) error {
+	log.Infof("Migrating revocation logs, might take a while...")
+
+	var (
+		err error
+
+		// finished is used to exit the for loop.
+		finished bool
+
+		// total is the number of total un-migrated records.
+		total uint64
+
+		// migrateCount is the total records migrated in current
+		// migration.
+		migrateCount float64
+	)
+
+	// First of all, read the number of un-migrated records.
+	err = kvdb.Update(db, func(tx kvdb.RwTx) error {
+		total, err = fetchTotalNumOfRecords(tx)
+		if err != nil {
+			return err
+		}
+
+		return nil
+	}, func() {})
+	if err != nil {
+		return err
+	}
+
+	log.Infof("Total un-migrated logs: %d", total)
```

Comment:
Ah ok , so this _does_ have some logging. However we never registered this logger so nothing is printed out atm.

Reply:
```diff
diff --git a/channeldb/log.go b/channeldb/log.go
index 42b55cc..658b53d 100644
--- a/channeldb/log.go
+++ b/channeldb/log.go
@@ -8,6 +8,7 @@ import (
        "github.com/lightningnetwork/lnd/channeldb/migration13"
        "github.com/lightningnetwork/lnd/channeldb/migration16"
        "github.com/lightningnetwork/lnd/channeldb/migration24"
+       "github.com/lightningnetwork/lnd/channeldb/migration29"
        "github.com/lightningnetwork/lnd/channeldb/migration_01_to_11"
        "github.com/lightningnetwork/lnd/kvdb"
 )
@@ -38,5 +39,6 @@ func UseLogger(logger btclog.Logger) {
        migration13.UseLogger(logger)
        migration16.UseLogger(logger)
        migration24.UseLogger(logger)
+       migration29.UseLogger(logger)
        kvdb.UseLogger(logger)
 }
```

Reply:
cool somehow missed, added.